### PR TITLE
V2 profiling

### PR DESCRIPTION
**Pull request by pmcollins**
Add profiling to auto instrumentation code path.

### CODE REVIEW

File: src/splunk_otel/distro.py [hunk 1]
```diff
@@ -41,15 +43,21 @@ def __init__(self):
...
+        self.set_profiling_env()
         self.configure_headers()
         self.set_server_timing_propagator()
 
     def set_env_defaults(self):
         for key, value in DEFAULTS.items():
             self.env.setdefault(key, value)
 
+    def set_profiling_env(self):
+        if self.env.is_true(SPLUNK_PROFILER_ENABLED):
```
Comment (johnbley) [inline src/splunk_otel/distro.py:57]: Explicitly state the default value of `false` here?

File: tests/fixtures/stacktraces.in.json [hunk 2]
```diff
@@ -0,0 +1,157 @@
+[
```
Comment (johnbley) [inline tests/fixtures/stacktraces.in.json:1]: 😄 love these types of tests!

File: tests/ott_profile.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        return [project_path(), "oteltest"]
+
+    def wrapper_command(self):
+        return "opentelemetry-instrument"
+
+    def on_start(self):
+        pass
+
+    def on_stop(self, tel: Telemetry, stdout: str, stderr: str, returncode: int):
+        assert num_logs(tel)
```
Comment (johnbley) [inline tests/ott_profile.py:25]: Some other tiny spot check that this smells like a profiler log?  Anything at all?

**Review by johnbley — APPROVED** [review]

File: src/splunk_otel/distro.py (hunk 1 above)
Comment (pmcollins) [reply src/splunk_otel/distro.py]: Done, thx.

File: tests/ott_profile.py (hunk 3 above)
Comment (pmcollins) [reply tests/ott_profile.py]: Good idea. Done.

**Pull request merged**